### PR TITLE
[baremetal] Clear internal decrypted buffer after read

### PR DESCRIPTION
**Pull request by jarvte**
Clear memory from internal decrypted buffers with memset.

### CODE REVIEW

File: library/ssl_tls.c [hunk 1]
```diff
@@ -11119,6 +11119,10 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
     mbedtls_platform_memcpy( buf, ssl->in_offt, n );
     ssl->in_msglen -= n;
 
+    // FI mitigation: "delete" incoming data after it's copied to buffer
+    mbedtls_platform_memset(ssl->in_offt, 0, n);
+    mbedtls_platform_memset(ssl->in_msg, 0, n);
+
```
Comment (artokin) [inline library/ssl_tls.c]: Few lines above there is ` ssl->in_offt = ssl->in_msg;`, are we clearing the same buffer twice?

Comment (jarvte) [reply library/ssl_tls.c]: actually yes, good catch

Comment (jarvte) [reply library/ssl_tls.c]: fixed

**Comment by jarvte** [comment]
> I don't think this is a FI mitigation, but anyway LGTM

I'll fix the comment and commit msg

**Comment by jarlamsa** [comment]
2 approving reviews and passing CI -> ready for merge

**Pull request merged**